src/index.js

```diff
@@ -236,19 +230,63 @@ window.onload = () => {
         ]);
       }
     });
-    whenMapAndDataReady(ddb, map);
   });
+};
+
+// Reload data every five minutes
+const FIVE_MINUTES_IN_MS = 300000;
+const recursiveDataLoad = () => {
+  pageDraws++;
   loadDataOnPage();
+  setTimeout(recursiveDataLoad, FIVE_MINUTES_IN_MS);
+};
 
-  // Reload data every five minutes
-  const FIVE_MINUTES_IN_MS = 300000;
-  const recursiveDataLoad = () => {
-    pageDraws++;
-    loadDataOnPage();
-    setTimeout(recursiveDataLoad, FIVE_MINUTES_IN_MS);
-  };
+// Call only if ddb is updated.
+// Uses a setTimeout to queue a new macrotask
+const callIfUpdated = (callback, delay = 0) => {
```

Comment:
another application of the setTimeout hack explained earlier. Here it's combined with a guard against the ddb being updated, based on the timestamp. It's not perfect, since it isn't actually comparing the actual data. We could get fancy and actually track whether each data source has changed, depending on what is needed for the chart.